#167756648 fix api bugs and add more modification

amkayondo opened the pull request:
#### What does this PR do?
fix api bugs and add more modification
- add more validations to endpoints 
- add validations for spaces and special characters
- add validations for special 3e3 characters on creating trips
- remove id and isAdmin in token for siginin and signup
- add a field for available seats to the app when the user books the seats available decrease and when deletes booking, the seats available increase
#### Description of Task to be completed?
fix api bugs and add more functionality
#### What are the relevant pivotal tracker stories?
[#167756648](https://www.pivotaltracker.com/story/show/167756648)
#### Screenshots (if appropriate)
![ghfghfgh](https://user-images.githubusercontent.com/46356108/62669291-5c35c380-b98f-11e9-9ece-545e460f0efe.PNG)
![Screenshot (10)](https://user-images.githubusercontent.com/46356108/62669292-5cce5a00-b98f-11e9-9d08-b8a66f4fba13.png)
![sddfsd](https://user-images.githubusercontent.com/46356108/62669294-5cce5a00-b98f-11e9-9077-b4a030d585b8.PNG)
![Capture](https://user-images.githubusercontent.com/46356108/62669295-5d66f080-b98f-11e9-9cd3-e84a2c144540.PNG)

amkayondo committed:
bug(user interface): Fix User Interface

- add more fields to siginup
    - location
    - Phone (edited)
- add a border to booking item
- remove the admin dash link from the user drop down menu
- add go to admin field on the index page
- add a footer to the page
- make the signin center and  wide

[finishes #167723968]

amkayondo committed:
chore (API ): fix Api Bugs and add more modification

- add more validations to endpoints -X
- add validations for spaces and special characters
- add validations for special 3e3 characters on creating trips
- remove id and isAdmin in token for siginin and signup
- add a field for available seats to app when the user books the seats available decrease and when deletes booking, the seats available increase

Oluwasegun-AA commented:
please change this error response to something more simple. i.e bad request, price should be interger between xxxxx and xxxxx
![image](https://user-images.githubusercontent.com/25525765/62680916-abdcb500-b9b8-11e9-8dfa-f4f901fcc6f0.png)

amkayondo committed:
bug(tests): Fix tests to run

-  create more tests for uodates made
-  update the data in line with the modifications
-  fix the response statuses
-  deploy the codebase to github
-  run tests on travis

[Finishes #167756648]
